t/app/controller/admin_reportextrafields.t

```diff
@@ -0,0 +1,311 @@
+            description => "this is a test description",
+            datatype_description => "hint here",
+            datatype => "string",
+        };
+        is_deeply $object->get_extra_fields, $extra_fields, 'new string field was added';
+        is $object->cobrand, 'tester', 'Correct cobrand set';
+        is $object->language, undef, 'Correct language set';
+
+        $mech->get_ok("/admin/reportextrafields/" . $object->id);
+        $mech->submit_form_ok( { with_fields => {
+            "language" => "en-gb",
+            "metadata[1].order" => "2",
+            "metadata[1].code" => "list_test",
+            "metadata[1].required" => undef,
+            "metadata[1].notice" => "",
+            "metadata[1].description" => "this field is a list",
+            "metadata[1].datatype_description" => "",
+            "metadata[1].datatype" => "list",
+            "metadata[1].values[0].key" => "key1",
+            "metadata[1].values[0].name" => "name1",
+        }});
+
+        push @$extra_fields, {
+            order => "2",
+            code => "list_test",
+            required => "false",
+            variable => "true",
+            description => "this field is a list",
+            datatype_description => "",
+            datatype => "singlevaluelist",
+            values => [
+                { name => "name1", key => "key1" },
+            ]
+        };
+        $object->discard_changes;
+        is_deeply $object->get_extra_fields, $extra_fields, 'new list field was added';
+        is $object->language, "en-gb", "Correct language was set";
+
+        $mech->get_ok("/admin/reportextrafields/" . $object->id);
+        $mech->submit_form_ok( { with_fields => {
+            "metadata[1].values[1].key" => "key2",
+            "metadata[1].values[1].name" => "name2",
+        }});
+
+        push @{$extra_fields->[1]->{values}}, { name => "name2", key => "key2" };
+        $object->discard_changes;
+        is_deeply $object->get_extra_fields, $extra_fields, 'options can be added to list field';
+    };
+
+    subtest 'Fields appear on /report/new' => sub {
+        $mech->get_ok("/report/new?longitude=-1.351488&latitude=51.847235&category=" . $contact->category);
+        $mech->content_contains("this is a test description");
+        $mech->content_contains("this field is a list");
+    };
+};
+
+FixMyStreet::override_config {
+    ALLOWED_COBRANDS => [ { 'tester' => '.' } ],
+    MAPIT_TYPES => [ 'UTA' ],
+    MAPIT_URL => 'http://mapit.uk/',
+    LANGUAGES => [ 'de,German,de_DE' ]
+}, sub {
+    subtest 'Language-specific fields are missing from /report/new for other language' => sub {
+        $mech->get_ok("/report/new?longitude=-1.351488&latitude=51.847235&category=" . $contact->category);
+        $mech->content_lacks("this is a test description");
+        $mech->content_lacks("this field is a list");
+    };
+};
+
+FixMyStreet::override_config {
+    ALLOWED_COBRANDS => [ { 'default' => '.' } ],
```

Comment:
As the default is 0, this is testing the same thing as the next test, really. I guess there should be a cobrand that is set to 1 but with a different name to actually test this.